### PR TITLE
SwiftUI PoC

### PR DESCRIPTION
**Pull request by cobbal**
I've been using Xcode 12 to develop this, which is why bitrise is unhappy I think.

[ignore-commit-lint]

### CODE REVIEW

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/ubyoirt) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fubyoirt)
teacher: [Link](https://rebrand.ly/kbbcy91) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fkbbcy91)
parent: [Link](https://rebrand.ly/ny4w1yn) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fny4w1yn)

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a reference to a jira ticket. For example: `refs: MBL-10023`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add which apps this change affects. Example: `affects: Teacher, Student` or `affects: none`
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a release note. If no release note is wanted, use `none`. Example: `release note: Fixed a bug that prevented users from enjoying the app.`
  </td>
    </tr>
  </tbody>
</table>



<details><summary>:x: JS tests</summary>


    Summary of all failing tests
     FAIL  src/modules/dashboard/__tests__/Dashboard.test.js (5.413s)
      ● Dashboard › render
    
        expect(received).toMatchSnapshot()
    
        Snapshot name: `Dashboard render 1`
    
        - Snapshot  - 1
        + Received  + 1
    
        @@ -95,11 +95,11 @@
                              undefined,
                            ],
                          ]
                        }
                      >
        -               See All
        +               All Courses
                      </Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <item>
    
          159 |       <Dashboard {...defaultProps} />
          160 |     ).toJSON()
        > 161 |     expect(tree).toMatchSnapshot()
              |                  ^
          162 |   })
          163 | 
          164 |   it('render while pending', () => {
    
          at Object.<anonymous> (src/modules/dashboard/__tests__/Dashboard.test.js:161:18)
    
      ● Dashboard › render while pending
    
        expect(received).toMatchSnapshot()
    
        Snapshot name: `Dashboard render while pending 1`
    
        - Snapshot  - 1
        + Received  + 1
    
        @@ -95,11 +95,11 @@
                              undefined,
                            ],
                          ]
                        }
                      >
        -               See All
        +               All Courses
                      </Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <item>
    
          166 |       <Dashboard {...defaultProps} pending={1} />
          167 |     ).toJSON()
        > 168 |     expect(tree).toMatchSnapshot()
              |                  ^
          169 |   })
          170 | 
          171 |   it('render without favorite courses', () => {
    
          at Object.<anonymous> (src/modules/dashboard/__tests__/Dashboard.test.js:168:18)
    
      ● Dashboard › render without favorite courses
    
        expect(received).toMatchSnapshot()
    
        Snapshot name: `Dashboard render without favorite courses 1`
    
        - Snapshot  - 1
        + Received  + 1
    
        @@ -95,11 +95,11 @@
                              undefined,
                            ],
                          ]
                        }
                      >
        -               See All
        +               All Courses
                      </Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <item>
    
          173 |       <Dashboard {...defaultProps} courses={[]} totalCourseCount={3} />
          174 |     ).toJSON()
        > 175 |     expect(tree).toMatchSnapshot()
              |                  ^
          176 |   })
          177 | 
          178 |   it('render without courses and *zero* total courses', () => {
    
          at Object.<anonymous> (src/modules/dashboard/__tests__/Dashboard.test.js:175:18)
    
      ● Dashboard › render without courses and *zero* total courses
    
        expect(received).toMatchSnapshot()
    
        Snapshot name: `Dashboard render without courses and *zero* total courses 1`
    
        - Snapshot  - 1
        + Received  + 1
    
        @@ -95,11 +95,11 @@
                              undefined,
                            ],
                          ]
                        }
                      >
        -               See All
        +               All Courses
                      </Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <item>
    
          180 |       <Dashboard {...defaultProps} courses={[]} concludedCourses={[]} totalCourseCount={0} />
          181 |     ).toJSON()
        > 182 |     expect(tree).toMatchSnapshot()
              |                  ^
          183 |   })
          184 | 
          185 |   it('select profile', () => {
    
          at Object.<anonymous> (src/modules/dashboard/__tests__/Dashboard.test.js:182:18)
    
      ● Dashboard › Doesnt render groups when in the teacher app
    
        expect(received).toMatchSnapshot()
    
        Snapshot name: `Dashboard Doesnt render groups when in the teacher app 1`
    
        - Snapshot  - 1
        + Received  + 1
    
        @@ -95,11 +95,11 @@
                              undefined,
                            ],
                          ]
                        }
                      >
        -               See All
        +               All Courses
                      </Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <item>
    
          299 |       <Dashboard {...defaultProps} />
          300 |     ).toJSON()
        > 301 |     expect(tree).toMatchSnapshot()
              |                  ^
          302 |     App.setCurrentApp('student')
          303 |   })
          304 | 
    
          at Object.<anonymous> (src/modules/dashboard/__tests__/Dashboard.test.js:301:18)
    
      ● Dashboard › renders course invites
    
        expect(received).toMatchSnapshot()
    
        Snapshot name: `Dashboard renders course invites 1`
    
        - Snapshot  - 1
        + Received  + 1
    
        @@ -579,11 +579,11 @@
                              undefined,
                            ],
                          ]
                        }
                      >
        -               See All
        +               All Courses
                      </Text>
                    </View>
                  </TouchableOpacity>
                </View>
                <item>
    
          391 |       <Dashboard {...props} />
          392 |     )
        > 393 |     expect(tree).toMatchSnapshot()
              |                  ^
          394 |     App.setCurrentApp(currentApp)
          395 |   })
          396 | 
    
          at Object.<anonymous> (src/modules/dashboard/__tests__/Dashboard.test.js:393:18)
    
    
    Snapshot Summary
     › 6 snapshots failed from 1 test suite. Inspect your code changes or run `yarn run test:ci -u` to update them.
    
    Test Suites: 1 failed, 266 passed, 267 total
    Tests:       6 failed, 2244 passed, 2250 total
    Snapshots:   6 failed, 512 passed, 518 total
    Time:        317.321s
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Store/Store.swift [hunk 1]
```diff
@@ -179,6 +180,13 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
         }
     }
 
+    @objc
+    public func controllerWillChangeContent(_ controller: NSFetchedResultsController<NSFetchRequestResult>) {
+        if #available(iOSApplicationExtension 13.0, *) {
+            objectWillChange.send()
+        }
+    }
```
Comment (vanwagonet) [inline Core/Core/Store/Store.swift:188]: if you move this down into the extension, you shouldn't need the second `if #available`

Comment (cobbal) [reply Core/Core/Store/Store.swift:188]: Unfortunately extensions and `@objc` don't get along

Comment (vanwagonet) [reply Core/Core/Store/Store.swift:188]: oh, right.

Comment (vanwagonet) [reply Core/Core/Store/Store.swift:188]: maybe an `@available` on the func then?

**Pull request closed without merging**